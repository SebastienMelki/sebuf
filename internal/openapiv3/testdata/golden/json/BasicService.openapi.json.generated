{"components":{"schemas":{"CreateUserRequest":{"description":"Simple request message","properties":{"email":{"description":"User email","type":"string"},"name":{"description":"User name","type":"string"}},"type":"object"},"FieldViolation":{"description":"FieldViolation describes a single validation error for a specific field.","properties":{"description":{"description":"Human-readable description of the validation violation (e.g., 'must be a valid email address', 'required field missing')","type":"string"},"field":{"description":"The field path that failed validation (e.g., 'user.email' for nested fields). For header validation, this will be the header name (e.g., 'X-API-Key')","type":"string"}},"required":["field","description"],"type":"object"},"GetUserRequest":{"description":"Get user request","properties":{"id":{"description":"User ID to retrieve","type":"string"}},"type":"object"},"User":{"description":"Simple response message","properties":{"email":{"description":"User email","type":"string"},"id":{"description":"User ID","type":"string"},"name":{"description":"User name","type":"string"}},"type":"object"},"ValidationError":{"description":"ValidationError is returned when request validation fails. It contains a list of field violations describing what went wrong.","properties":{"violations":{"description":"List of validation violations","items":{"$ref":"#/components/schemas/FieldViolation"},"type":"array"}},"required":["violations"],"type":"object"}}},"info":{"title":"BasicService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/BasicService/SimpleMethod":{"post":{"description":"Simple method without HTTP config","operationId":"SimpleMethod","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/CreateUserRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"SimpleMethod","tags":["BasicService"]}},"/configured":{"post":{"description":"Method with only path config","operationId":"ConfiguredMethod","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/GetUserRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"ConfiguredMethod","tags":["BasicService"]}}}}