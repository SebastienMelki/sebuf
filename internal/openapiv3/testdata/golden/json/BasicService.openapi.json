{"components":{"schemas":{"CreateUserRequest":{"description":"Simple request message","properties":{"email":{"description":"User email","type":"string"},"name":{"description":"User name","type":"string"}},"type":"object"},"GetUserRequest":{"description":"Get user request","properties":{"id":{"description":"User ID to retrieve","type":"string"}},"type":"object"},"User":{"description":"Simple response message","properties":{"email":{"description":"User email","type":"string"},"id":{"description":"User ID","type":"string"},"name":{"description":"User name","type":"string"}},"type":"object"}}},"info":{"title":"BasicService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/BasicService/SimpleMethod":{"post":{"description":"Simple method without HTTP config","operationId":"SimpleMethod","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/CreateUserRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"SimpleMethod","tags":["BasicService"]}},"/configured":{"post":{"description":"Method with only path config","operationId":"ConfiguredMethod","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/GetUserRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"ConfiguredMethod","tags":["BasicService"]}}}}